pkg/opensearch/response_parser.go

```diff
@@ -1012,7 +1016,8 @@ func (rp *responseParser) processAggregationDocs(esAgg *simplejson.Json, aggDef
 		for _, e := range fields {
 			for _, propKey := range propKeys {
 				if e.Name == propKey {
-					e.Append(props[propKey])
+					value := props[propKey]
+					e.Append(&value)
```

Comment:
this bit of the code used to build the results the same way as the frontend flow (building a table), but was [refactored to use data frames](https://github.com/grafana/opensearch-datasource/pull/186) instead. that refactoring was done by following elasticsearch's pr (which had the same bug that this PR is addressing). so it probably wasn't being executed from that point onwards. but we didn't notice because the backend flow would've only been run for alerts or in explore and it would've only been an issue if someone was using multiple group by terms.